_spikes-poc/function-fs-access/functions/test-fs-access/v1.js

```diff
@@ -0,0 +1,189 @@
+'use strict'
+
+const functionName = 'test-fs-access'
+
+const os = require('os')
+const fs = require('fs-extra')
+const path = require('path')
+
+const archiver = require('archiver')
+const checkDisk = require('check-disk-space')
+
+let logger
+
+function createLargeFile (dir, filePrefix) {
+  return new Promise(async (resolve, reject) => {
+    const fname = `${dir}${path.sep}${filePrefix}`
+    const stream = fs.createWriteStream(fname, { mode: 0o600 })
+    for (let i = 0; i < 10000000; i++) {
+      if (!stream.write(`Line of random text ${i}\n`)) {
+        // Will pause every 16384 iterations until `drain` is emitted
+        await new Promise(resolve => stream.once('drain', resolve))
+      }
+    }
+    stream.end()
+    stream.on('finish', function () {
+      resolve(fname)
+    })
+    stream.on('error', function (error) {
+      reject(error)
+    })
+  })
+}
+
+/**
+ * Create a unique directory in the system's temp dir
+ * @param prefix
+ * @return {*}
+ */
+async function createTmpDir (prefix) {
+  return fs.mkdtemp(`${os.tmpdir()}${path.sep}${prefix}`)
+}
+
+function readBack (fileName) {
+  return new Promise((resolve, reject) => {
+    const stream = fs.createReadStream(fileName)
+    let data = ''
+    let linesWanted = 10
+    let lines
+
+    stream.on('data', function (chunk) {
+      data += chunk
+      lines = data.split('\n')
+      if (lines.length > linesWanted) {
+        lines.length = linesWanted
+        stream.destroy()
+      }
+    })
+
+    stream.on('error', function (error) {
+      console.error(`Error reading file ${fileName}: ${error.message}`)
+      reject(error)
+    })
+
+    stream.on('end', function () {
+      logger(`End event: Contents of ${fileName}: ${lines.join('\n')}`)
+      resolve()
+      data = null
```

Comment:
@jon-shipley is setting data to null necessary?

Reply:
Probably not, but it forces memory to be freed.